plugins/nirmata/api_token.go

```diff
@@ -0,0 +1,88 @@
+package nirmata
+
+import (
+	"context"
+
+	"github.com/1Password/shell-plugins/sdk"
+	"github.com/1Password/shell-plugins/sdk/importer"
+	"github.com/1Password/shell-plugins/sdk/provision"
+	"github.com/1Password/shell-plugins/sdk/schema"
+	"github.com/1Password/shell-plugins/sdk/schema/credname"
+	"github.com/1Password/shell-plugins/sdk/schema/fieldname"
+)
+
+func APIToken() schema.CredentialType {
+	return schema.CredentialType{
+		Name:          credname.APIToken,
+		DocsURL:       sdk.URL("https://nirmata.com/docs/api_token"),               // TODO: Replace with actual URL
+		ManagementURL: sdk.URL("https://console.nirmata.com/user/security/tokens"), // TODO: Replace with actual URL
+		Fields: []schema.CredentialField{
+			{
+				Name:                fieldname.Token,
+				MarkdownDescription: "Token used to authenticate to Nirmata.",
+				Secret:              true,
+				Composition: &schema.ValueComposition{
+					Length: 116,
+					Charset: schema.Charset{
+						Lowercase: true,
+						Digits:    true,
+					},
+				},
+			},
+			{
+				Name:                fieldname.Email,
+				MarkdownDescription: "Email address registered in Nirmata.",
+			},
+			{
+				Name:                fieldname.Address,
+				MarkdownDescription: "Url address of Nirmata[https://nirmata.io].",
+			},
+		},
+		DefaultProvisioner: provision.EnvVars(defaultEnvVarMapping),
+
+		Importer: importer.TryAll(
+			importer.TryEnvVarPair(defaultEnvVarMapping),
+			TryNirmataConfigFile(),
+		)}
+}
+
+var defaultEnvVarMapping = map[string]sdk.FieldName{
+	"NIRMATA_TOKEN": fieldname.Token,
+	"NIRMATA_URL":   fieldname.URL,
+}
```

Comment:
I see we import and require an email, but only provision the token and URL. Is the email something that is required as well?

Otherwise, let's consider making it optional, or, alternatively, omitting it from the importer.

Reply:
yup done!